grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaChannel.java

```diff
@@ -101,25 +94,27 @@
         ctx.logBuilder().serializationFormat(serializationFormat);
         ctx.logBuilder().deferRequestContent();
         ctx.logBuilder().deferResponseContent();
+
+        final ClientOptions options = options();
+        final int maxOutboundMessageSizeBytes = options.get(GrpcClientOptions.MAX_OUTBOUND_MESSAGE_SIZE_BYTES);
+        final int maxInboundMessageSizeBytes = options.get(GrpcClientOptions.MAX_INBOUND_MESSAGE_SIZE_BYTES);
+        final boolean unsafeWrapResponseBuffers = options.get(GrpcClientOptions.UNSAFE_WRAP_RESPONSE_BUFFERS);
+
         return new ArmeriaClientCall<>(
                 ctx,
                 params.endpointGroup(),
                 httpClient,
                 req,
                 method,
-                options().getOrElse(GrpcClientOptions.MAX_OUTBOUND_MESSAGE_SIZE_BYTES,
-                                    ArmeriaMessageFramer.NO_MAX_OUTBOUND_MESSAGE_SIZE),
-                options().getOrElse(
-                        GrpcClientOptions.MAX_INBOUND_MESSAGE_SIZE_BYTES,
-                        options().getOrElse(
-                                ClientOption.MAX_RESPONSE_LENGTH,
-                                (long) DEFAULT_MAX_INBOUND_MESSAGE_SIZE).intValue()),
+                maxOutboundMessageSizeBytes,
+                maxInboundMessageSizeBytes > 0 ? maxInboundMessageSizeBytes
+                                               : Ints.saturatedCast(options.maxResponseLength()),
```

Comment:
Just to confirm is saturated cast the behavior change mentioned in the notes?

LGTM

Reply:
No. The previous behavior was this:

```java
                options().getOrElse(
                        GrpcClientOptions.MAX_INBOUND_MESSAGE_SIZE_BYTES,
                        options().getOrElse(
                                ClientOption.MAX_RESPONSE_LENGTH,
                                (long) DEFAULT_MAX_INBOUND_MESSAGE_SIZE).intValue()),
```

We were doing saturated cast already.

The difference if that `DEFAULT_MAX_INBOUND_MESSAGE_SIZE` is not used anymore.

Reply:
Ah thanks for clarifying - the new options API opens up this nice tweak :)